specs/wp-coupons-spec.js

```diff
@@ -0,0 +1,244 @@
+/** @format */
+
+import config from 'config';
+import assert from 'assert';
+
+import * as driverManager from '../lib/driver-manager.js';
+import * as driverHelper from '../lib/driver-helper.js';
+import * as dataHelper from '../lib/data-helper.js';
+
+import StartPage from '../lib/pages/signup/start-page.js';
+
+import AboutPage from '../lib/pages/signup/about-page.js';
+import PickAPlanPage from '../lib/pages/signup/pick-a-plan-page.js';
+import CreateYourAccountPage from '../lib/pages/signup/create-your-account-page.js';
+
+import FindADomainComponent from '../lib/components/find-a-domain-component.js';
+import SecurePaymentComponent from '../lib/components/secure-payment-component.js';
+import SidebarComponent from '../lib/components/sidebar-component';
+
+import SignUpStep from '../lib/flows/sign-up-step';
+import LoginFlow from '../lib/flows/login-flow';
+import StatsPage from '../lib/pages/stats-page';
+import PlansPage from '../lib/pages/plans-page';
+
+const mochaTimeOut = config.get( 'mochaTimeoutMS' );
+const startBrowserTimeoutMS = config.get( 'startBrowserTimeoutMS' );
+const screenSize = driverManager.currentScreenSize();
+const signupInboxId = config.get( 'signupInboxId' );
+const host = dataHelper.getJetpackHost();
+const locale = driverManager.currentLocale();
+const passwordForTestAccounts = config.get( 'passwordForNewTestSignUps' );
+
+let driver;
+
+before( async function() {
+	this.timeout( startBrowserTimeoutMS );
+	driver = await driverManager.startBrowser();
+} );
+
+describe( `[${ host }] Sign Up  (${ screenSize }, ${ locale })`, function() {
+	this.timeout( mochaTimeOut );
+
+	describe( 'Sign up for a non-blog site on a premium paid plan through main flow in USD currency using a coupon @parallel @visdiff', function() {
```

Comment:
FYI this test will fail when the GeoIP-mapped currency is not USD, because

1. The expected currency mismatched, as we discussed in https://github.com/Automattic/wp-e2e-tests/pull/1758
1. There are currencies that don't use decimal digits, so expecting the coupon-applied value to be exactly 0.99 times the original is not correct.

I have a candidate fix for the above two issues in https://github.com/Automattic/wp-e2e-tests/pull/1758/ .

Reply:
Merged #1758